src/NuGet.Clients/NuGet.VisualStudio.Common/NuGetProjectDependencyVersionLookup.cs

```diff
@@ -0,0 +1,105 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using NuGet.Packaging;
+using NuGet.Packaging.Core;
+using NuGet.ProjectManagement;
+using NuGet.ProjectManagement.Projects;
+using NuGet.Versioning;
+
+namespace NuGet.PackageManagement.VisualStudio
+{
+    public class NuGetProjectDependencyVersionLookup
```

Comment:
What the client does right now is to load the installed package identities from the config file (either project.json or packages.config, or csproj) and use them as a base for loading the UI. (Maybe I'm understanding something incorrectly?) What I'm trying to accomplish is updating those already loaded references with the resolved version. If this type is not specific to build integrated projects it would read the config file again (for packages.config projects) and try to update the values with the ones it already has.

At first I thought it would be a good idea to make this change from where the identities are loaded (for example `BuildIntegratedPackageReference:54`), but it seemed that going that direction was a more involved change on how package references are built for this type of projects.